src/track/trackrecord.cpp

```diff
@@ -45,14 +45,21 @@ bool TrackRecord::updateGlobalKey(
 UpdateResult TrackRecord::updateGlobalKeyText(
         const QString& keyText,
         track::io::key::Source keySource) {
-    Keys keys = KeyFactory::makeBasicKeysFromText(keyText, keySource);
-    if (keys.getGlobalKey() == track::io::key::INVALID) {
+    // Try to parse the input as a key.
+    mixxx::track::io::key::ChromaticKey newKey =
+            KeyUtils::guessKeyFromText(keyText);
+    if (newKey == mixxx::track::io::key::INVALID) {
+        // revert if we can't guess a valid key from it
         return UpdateResult::Rejected;
     }
-    if (m_keys.getGlobalKey() == keys.getGlobalKey()) {
+
+    // If the new key is the same as the old key, reject the change.
+    if (m_keys.getGlobalKey() == newKey) {
         return UpdateResult::Unchanged;
     }
-    setKeys(keys);
+
+    Keys newKeys = KeyFactory::makeBasicKeys(newKey, keySource);
```

Comment:
won't this normalize user key data? eg shouldn't this be `makeBasicKeysKeepText`?

Reply:
Yes. We can discuss about this. 

For now I have kept the original behaviour that user edits are checked and normalized to make sure the string written to the file metadata is useful for other tools. So in case the user enters a string that Mixxx understands, but violates the id3 format it is corrected. 

makeBasicKeysKeepText() is used to not touch a string that is already written into the file metadata. Checking and modifying it is not the busyness of Mixxx IMHO.

I can imagine a use case where the user want to manual edit a non standard string but there is a risk to mess the strings up. Without a special use case I prefere not to open it. 

What do you think? 

I think I will add a comment.

Reply:
Oh I forget. This PR is only the refactoring PR without functional changes. The considerations above are useful for the follow up PR.

Reply:
Well the concern I had when I commented this initially was that the previous behavior was more relaxed and did not normalize. Since the method is called `TrackRecord::updateGlobalKeyText` I thought this code is not exclusive to user-edits but also gets run on import of pre-existing metadata which is why I thought that shouldn't be normalized. If this code is only run on user-edit within mixxx, normalizing would be the correct thing to do IMO.

Reply:
I have renamed the function. It runs in both cases fixing this in the "read from file metadata" case is the topic of the follow up PR. The intention of this PR is not to change the behavior.

Reply:
I'm sorry. I've misunderstood the code. LGTM.